src/include/sof/lib/pm_runtime.h

```diff
@@ -93,6 +93,16 @@ void pm_runtime_put(enum pm_runtime_context context, uint32_t index);
  */
 void pm_runtime_put_sync(enum pm_runtime_context context, uint32_t index);
 
+/**
+ * \brief Reports state of the power managed resource.
+ *
+ * @param context Type of power management context.
+ * @param index Index of the resource.
+ *
+ * @return true if the resource is active, false otherwise.
+ */
+bool pm_runtime_active(enum pm_runtime_context context, uint32_t index);
```

Comment:
Can we say pm_runtime_is_active() here, it makes the API usage flow better in English.

Reply:
Renamed, however I just followed the naming from the pm-runtime documented at kernel.org. There is `set_active`/`active` setter/getter pair (well, there is also `irq_safe`/`is_irq_safe` in another place).

Important note: while reading the documentation at kernel.org again, I realized that DSP D0 state should be enforced by pm_runtime_enable/disable API, not put/get, so I added them as well and changed the DSP D0 implementation in cAVS a bit. I apologize for adding a new piece of the code to the reviewed patch, but enable/disable seem much more appropriate as an API to implement the last commit that had been already there (Btw I have a couple more use cases for enable/disable rather than put/get in the queue).